ios-base/DroidKaigi 2020/Announcements/Models/AnnouncementsDataProvider.swift

```diff
@@ -0,0 +1,62 @@
+import ios_combined
+import RxSwift
+
+protocol AnnouncementsDataProviderProtocol {
+    func fetch() -> Single<[Announcement]>
+}
+
+final class AnnouncementsDataProvider: AnnouncementsDataProviderProtocol {
+    enum Transformer {
+        static let dateFormatter: DateFormatter = {
+            let formatter = DateFormatter()
+            formatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ssXXX"
+            return formatter
+        }()
+
+        static func transform(_ response: String) -> Announcement.Type_? {
+            switch response {
+            case "notification":
+                return .notification
+            case "alert":
+                return .alert
+            case "feedback":
+                return .feedback
+            default:
+                return .none
+            }
+        }
+
+        static func transform(_ response: AnnouncementResponse) -> Announcement? {
+            guard let publishedAt = dateFormatter.date(from: response.publishedAt)?.timeIntervalSince1970 else {
+                return .none
+            }
+            guard let type = transform(response.type) else {
+                return .none
+            }
+
+            return Announcement(
+                id: response.id_,
+                title: response.title,
+                content: response.content,
+                publishedAt: publishedAt,
+                type: type
+            )
+        }
+    }
```

Comment:
In session, I used Kotlin's transformer class such as this https://github.com/DroidKaigi/conference-app-2020/blob/master/ios-base/DroidKaigi%202020/Session/Models/SessionDataProvider.swift#L8.
If it is possible to implement with Kotlin, please implement with it 🙏

Reply:
I have read Android app's code. The announcement model conversion process is API model to entity, entity to model. But, iOS app's code doesn't implement entity layer. So, I think not useful to share announcement conversion logic between Kotlin and Swift.
Should I implement conversion logic in Kotlin?

Reply:
Why I use the Kotlin model on iOS, I want to share the methods that classes like Session have. 👀

Reply:
If you have any opinion, please tell me 🙇

Reply:
I understand that reason. I am using Announcement model in iOS.
Android doesn't use Response.toModel method implemented in Kotlin, right? I think to be shared Model. But I think can't be shared conversion logic API response to model. Because iOS doesn't implement entity layer now.

すいません、英語で伝えられる文章を書ききれる自信がないため、日本語で失礼します 🙏 
その理由は理解できています。AndroidはKotlinで実装された `Response.toModel` を利用していない（つまり、変換ロジックはAndroid/iOSで共用されていない）ですが、変換ロジックをiOSで実装したほうがよいですか？先に記述したように、iOSではEntity Layerを用意しておらず、Androidの変換プロセス（API Response -> entity -> model）と一致しないので、現状は共通化が困難だと考えています。

Reply:
変換先のモデルがKotlinのモデルであれば、変換はSwiftでいいと思います！！(自分がiOSで使う変換ロジックをKotlinで書いたのは、単にKotlinに慣れていたからです)

Reply:
Thanks!! I'll implement conversion logic in Swift 😄